backend/ticketvise/views/api/ticket.py

```diff
@@ -245,3 +248,37 @@ def perform_create(self, serializer):
 
         for file in self.request.FILES.getlist('files'):
             TicketAttachment(ticket=ticket, file=file).save()
+
+
+class TicketSharedWithRetrieveSerializer(ModelSerializer):
+    shared_with = UserSerializer(many=True, read_only=True)
+
+    class Meta:
+        model = Ticket
+        fields = ["shared_with"]
+
+
+class TicketSharedWithUpdateSerializer(ModelSerializer):
+    class Meta:
+        model = Ticket
+        fields = ["shared_with"]
+
+    def validate_shared_with(self, shared_with):
+        inbox = self.instance.inbox
+        for user in shared_with:
+            if not user.has_inbox(inbox) or user.is_assistant_or_coordinator(inbox):
+                raise ValidationError("This ticket cannot be shared with one of these users")
+        return shared_with
+
+
+class TicketSharedAPIView(UserIsTicketAuthorOrInboxStaffMixin, RetrieveUpdateAPIView):
```

Comment:
Gebruik je hier het Retrieve gedeelte ook van de APIView?

Reply:
`RetrieveUpdateAPIView` wordt gebruikt ja. Beide views worden gebruikt, alleen worden er met behulp van het overschrijven van de `get_serializer_class()` verschillende serializers gebruikt. Dit heeft te maken met het zien van een `shared_with` als user of enkel een id